go-controller/pkg/config/config.go

```diff
@@ -1735,10 +1768,31 @@ func completeGatewayConfig(allSubnets *configSubnets) error {
 	if err != nil || !utilnet.IsIPv6(v6IP) {
 		return fmt.Errorf("invalid gateway v6 join subnet specified, subnet: %s: error: %v", Gateway.V6JoinSubnet, err)
 	}
-
 	allSubnets.append(configSubnetJoin, v4JoinCIDR)
 	allSubnets.append(configSubnetJoin, v6JoinCIDR)
 
+	//validate v4 and v6 masquerade subnets
+	v4MasqueradeIP, v4MasqueradeCIDR, err := net.ParseCIDR(Gateway.V4MasqueradeSubnet)
+	if err != nil || utilnet.IsIPv6(v4MasqueradeIP) {
+		return fmt.Errorf("invalid gateway v4 masquerade subnet specified, subnet: %s: error: %v", Gateway.V4MasqueradeSubnet, err)
+	} else {
+		if err = allocateV4MasqueradeIPS(v4MasqueradeIP, masqueradeIPs); err != nil {
+			return fmt.Errorf("unable to allocate V4MasqueradeIPs: %s", err)
+		}
+	}
```

Comment:
NIT: you can simplify the syntax above by omitting the `else` statement, since there's a return for the if condition above, and indent the else block to the left :)